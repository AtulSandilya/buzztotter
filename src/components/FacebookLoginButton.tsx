import * as React from "react";
import { Component, PropTypes } from 'react';

import { LoginButton, AccessToken } from 'react-native-fbsdk';

interface FacebookLoginButtonProps {
  loginDispatch?(token: string): void;
  logoutDispatch?(): void;
}

const FacebookLoginButton: React.StatelessComponent<FacebookLoginButtonProps> = ({loginDispatch, logoutDispatch}) => (
  <LoginButton
    readPermissions={['public_profile', 'user_friends', 'email', 'user_birthday']}
    onLoginFinished={(error, result) => {
      if(error){
        alert("Login error: " + result.error);
      } else if (result.isCancelled){
        alert("Login cancelled");
      } else {
        AccessToken.getCurrentAccessToken().then(
          (data) => {
            if(loginDispatch){
              loginDispatch(data.accessToken.toString());
            }
          }
        )
      }
    }}
    onLogoutFinished={() => {
      if(logoutDispatch){
        logoutDispatch();
      }
    }}
  />
)

export default FacebookLoginButton;
