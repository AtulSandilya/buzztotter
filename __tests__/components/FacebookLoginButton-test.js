import 'react-native';
import React from 'react';
import FacebookLoginButton from '../../build/components/FacebookLoginButton';

import renderer from 'react-test-renderer';

describe('Facebook Login Button', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <FacebookLoginButton />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
