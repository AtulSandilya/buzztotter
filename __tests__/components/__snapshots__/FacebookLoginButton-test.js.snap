exports[`Facebook Login Button renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#3b5998",
          "borderRadius": 5,
        },
      ],
    ]
  }
  testID={undefined}>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "#3b5998",
          "borderRadius": 5,
        },
        Object {
          "paddingHorizontal": 15,
          "paddingVertical": 5,
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 24,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      ï‚š
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }>
      Login with Facebook
    </Text>
  </View>
</View>
`;
